peer.go

```diff
@@ -21,6 +21,22 @@ import (
 //   sha2-256 multihash of the public key.
 const MaxInlineKeyLength = 42
 
+// ErrEmptyPeerID is an error for empty peer ID
+type ErrEmptyPeerID struct {
+	message string
+}
+
+// NewErrEmptyPeerID creates an instance of ErrEmptyPeerID
+func NewErrEmptyPeerID() *ErrEmptyPeerID {
+	return &ErrEmptyPeerID{
+		message: "empty peer ID",
+	}
+}
+
+func (e *ErrEmptyPeerID) Error() string {
+	return e.message
+}
+
```

Comment:
did you ask for something like
```
const ErrEmptyPeerID = "empty peer ID"
```
This is inspired from https://medium.com/@sebdah/go-best-practices-error-handling-2d15e1f0c5ee

Reply:
Just saw this https://github.com/libp2p/go-libp2p-swarm/blob/a00fa8aa37a812f404bfa9b9d3dccc82a66029cf/swarm_dial.go#L30-L44
Let me add changes